internal/dag/dot.go

```diff
@@ -203,20 +203,27 @@ func (g *marshalGraph) writeBody(opts *DotOpts, w *indentWriter) {
 					Attrs:  make(map[string]string),
 				}
 
-				dotEdges = append(dotEdges, cycleDot(e, g))
+				dotEdges[e.Name] = cycleDot(e, g)
 				src = tgt
 			}
 		}
 	}
 
 	for _, e := range g.Edges {
-		dotEdges = append(dotEdges, e.dot(g))
+		// only add the edge if it's not been added as part of a cycle
```

Comment:
This might be desired here, but not because of cycles. I think what this might be doing is just removing duplicate edges because the graph has not been reduced.